Jas dev

JasonThomasData opened the pull request:
This PR does a few things:
- CSS issue with Doctor Surgery name sitting weird fixed.
- Changed the doctors helper function name so it's more obvious.
- Implemented button for user to find their address on click, but only available from root. Wasn't sure what to do with this button, so I justified it right with the other button. So far tested on Firefox, but I can not test any other browsers on this computer.

Hoping this doesn't merge auto.

JasonThomasData committed:
Changed function so it is more clear, easier to read

JasonThomasData committed:
Moved the surgery name out of the header in the doctors view so its left justified

JasonThomasData committed:
Adding a button to the landing page, only the root, to allow the user to locate themself using HTML5 geolocation

JasonThomasData committed:
Right justified get_location button, looks no good on left

JasonThomasData committed:
Merge branch 'master' of https://github.com/JasonThomasData/DocLink into jas_dev

Pulling down the latest version of master.

JasonThomasData committed:
Readded jQuery so I can test this button to geolocate user device

JasonThomasData committed:
Added js again, didnt work last time, this time its loading

JasonThomasData committed:
App now has a button on homepage that finds the device location

JasonThomasData committed:
App now has a button on homepage that finds the device location

JasonThomasData committed:
Arabic locale updated with button text, more obvious JS variable names

JasonThomasData committed:
Adding travis yml file so we can do CI

JasonThomasData committed:
Adding travis again, nothing happened last time

JasonThomasData commented:
Oh that's where the tests show up. That's nice! Cool, hopefully I've set up the yml file properly and it works. That's quite enough for one day I think.

equivalentideas committed:
Fix indentation so it's consistent

equivalentideas committed:
Test the full behaviour, that it removes "/" as well now

equivalentideas committed:
We're not using turbo links

equivalentideas commented:
> CSS issue with Doctor Surgery name sitting weird fixed.
> Changed the doctors helper function name so it's more obvious.

These two look good. In future, with PRs it's best to make it as small as possible by limiting each PR to one change. That way you can review each change by itself. For example if we decided we didn't want one of the changes in the PR, we'd have to make a new PR and cherry-pick out the commits we wanted. That gets real messy ;-) So ideally you'd do a PR for each of those dot points. Makes review super quick.

Also it means you can name the PR after what the change is :P

Looking at the third point now.

equivalentideas commented:
With 429368580e82b8b55ca7f663ccf149023d5a4d36 in future you can use `git commit --amend` to fix the first commit, rather than split it into two. Also there's stuff in that commit that isn't about adding JS.

Same with commit b0c8c91 , you can `--amend` the first commit.

equivalentideas commented on config/routes.rb:
```diff
@@ -8,6 +8,6 @@
     get "translate", to: "translations#index"
     get "about", to: "static#about"
-    root to: "doctors#search"
+    root to: "doctors#search_root"
```
I don't understand why this needs to change?

JasonThomasData replied:
Damn I accidentally pressed cancel! Argh. 

I did this so the button would only appear at root, since it seems confusing to have the find location button on the results page. 

I think this might make sense to keep it this way, since having the css rule visibility:hidden might be messy on the results page, which also uses the search view, cause I think it's going to mean there's a blank space.

The other option is to actually add the button to the page with jQuery.

JasonThomasData replied:
Now I remember why I did this. I cannot figure out how to send JS to be loaded on root page only. Checking for the / path won't work because there can be things added to the path at root. Cannot figure this out.

JasonThomasData replied:
Do you think it's acceptable to keep the `search_root` view with the button and get JS to reveal the button on load? 
Page-specific JS for this page confuses me, since there's nothing specific on the root page is there?

JasonThomasData replied:
Um, no I think that is a bad idea. Is there anyway to have page-specific JS on the root only @equivalentideas ??

equivalentideas replied:
I think a good, progressive enhancement approach would be to add the button to the page with jquery if the search form has a class like ‘search-with-locate-button’. That way browsers without js don't get it.

Then we just add that class to the form on pages where we want the locate button.

That’s quite flexible then, and it’s not dependent on the Controller/route which might change.

equivalentideas commented on app/views/doctors/_doctor_search_form.html.haml:
```diff
@@ -2,5 +2,4 @@
     = search_field_tag :q, @q, placeholder: t("doctor_search_form.placeholder")
-    = button_tag t("doctor_search_form.submit_text"), type: "submit", name: nil
-
+    = button_tag t("doctor_search_form.submit_text"), type: "submit", name: nil
```
Since this button only works if you have JS, we should probably only show it if you do (and if you're device has location capabilities, but that's probably a bit more complex).

I recon the best way to do that is add the button into the form with JS.

equivalentideas replied:
Sorry got confused an commented on this one. Is this meant to be on this commit?

equivalentideas commented on app/assets/stylesheets/base/_buttons.scss:
```diff
@@ -33,3 +33,9 @@
   }
 }
+
+#get_location{
```
Could above with a style to differentiate it from the submit button be clearer?

![screen shot 2016-05-27 at 6 14 11 pm](https://cloud.githubusercontent.com/assets/1239550/15602175/fce4c02a-2436-11e6-81f3-bcc73739d979.png)

JasonThomasData replied:
I thought font awesome would be useful to differentiate it, that was why I suggested that originally actually. Anyway we can do that later.

I don't really like grey personally but I like how it goes 100%. So I'll change that.

equivalentideas commented on app/views/doctors/search_root.html.haml:
```diff
@@ -0,0 +1,4 @@
+%header.site-header
+  %h1.site-title= t(:app_purpose)
+  = render "doctors/doctor_search_form"
+  = button_tag t("get_location.submit_text"), type: "submit", name: nil, id:'get_location'
```
Since this button only works if you have JS, we should probably only show it if you do (and if you're device has location capabilities, but that's probably a bit more complex).

I recon the best way to do that is add the button into the form with JS.

What you recon @JasonThomasData ?

Also it shouldn't have the type "submit" cause it doesn't submit anything.

JasonThomasData replied:
I've tried this now, and I don't think it will work, unless we can get JS to handle the language.
Since Rails is handling the language stuff, I think the best way to do this is to have it in rails and activate+reveal the button in JavaScript. Doing that with JS requires us to keep knowledge in two places potentially.

equivalentideas commented:
Nice one @JasonThomasData. I left a few comments with tweeks to the locate button thing. Otherwise all good 👍 

It works smooth in Chrome too ;-)

There's some messy commits in here with unrelated changes and duplicate names. Watch out for that ;-)

I'll assign this back to you know to resolve the stuff from the comments and then you can assign back to me to merge :D

equivalentideas commented on app/assets/javascripts/device_location.js:
```diff
@@ -0,0 +1,43 @@
+function showAddress(lat, lng){
+	alert(lat + ' ' + lng)
+}
```
Are you really into this tab indentation?

With Rails it's generally the convention to use 2 spaces for everything.

JasonThomasData replied:
It's generally OK in JS to use tabs for indentation. I mean it comes down to personal preference. There's a whole emotional conversation about it here. http://programmers.stackexchange.com/questions/57/tabs-versus-spaces-what-is-the-proper-indentation-character-for-everything-in-e

Since this is a rails project, I'll make the tabs 2 spaces for consistency.

JasonThomasData committed:
Changing JS to be two spaces indented

Since we're using Ruby, it's nice to have similar indentation across the project.

JasonThomasData commented:
Also, if I'm going to make some commits today, I might just go and add font awesome to the buttons. Thoughts on that? So make it so the geolocate button has a map marker icon, search button has a magnifying glass.

On the results page, make it so the numbers have little phone icons.

Is this a useful addition do you think?

equivalentideas commented:
> Is this a useful addition do you think?

Not really. Feel like tinkering compared to addressing stuff in this PR and getting this merged ;-)

JasonThomasData committed:
Returned to one search view, removed button, since we will use JS to add button

JasonThomasData commented:
Hi, I added some line comments but they disappeared.
I was commenting on why I did the search_view thing.
The reason I did that is I could not find a way to implement page-specific JS.
Having the button on the search form is no good, because that same view is used on the doctors results view.

I think the obvious way to do this now is have a placeholder div and append a button to it with jQuery, but only from root. 
The page-specific JS stuff confuses me, because as far as I know there's nothing unique about the root page is there?

JasonThomasData committed:
Added button with JavaScript. People who dont have JS will not have button

JasonThomasData committed:
Discovered problem with last commit. Button added with Rails, activate with JS

JasonThomasData committed:
Javascript reveals button and makes button active

JasonThomasData committed:
Changed styles for the Find my location button

JasonThomasData committed:
Style and toggle the button with the hidden attribute

This is more accessible and simpler code.
